include/system/analytics_manager.js

```diff
@@ -108,11 +136,184 @@ module.exports = function AnalyticsManagerModule(pb) {
     };
 
     /**
+     *
+     * @static
+     * @method buildRenderTask
+     * @param {String} key
+     * @param {Request} req
+     * @param {Object} session
+     * @param {Localization} ls
+     * @return {Function}
+     */
+    AnalyticsManager.prototype.buildRenderTask = function(key, req, session, ls) {
+        var site = this.site;
+        var timeout = this.timeout;
+
+        return function(callback) {
+            if (pb.log.isSilly()) {
+                pb.log.silly("AnalyticsManager: Rendering provider [%s] for URL [%s:%s]", keys[i], req.method, req.url);
+            }
+
+            //build context object for builder functions to have access to params
+            var opts = {
+                method: req.method,
+                url: req.url,
+                key: key,
+                callback: callback,
+                th: null,
+                start: null
+            };
+
+            //create a domain for the task to operate within
+            var d = domain.create();
+            d.run(function() {
+
+                //set the execution timeout.  We set it here and not outside of the
+                //domain to ensure that it gets a fair shot at running for the entire
+                //allotted timeout.  Setting it outside makes it subject to other lines
+                //of execution already in queue that will burn through the timeout
+                //before execution of this domain begins.
+                opts.start = new Date().getTime();
+                opts.th = setTimeout(AnalyticsManager.buildTimeoutHandler(opts), timeout);
```

Comment:
Note the comment. Putting it outside of the domain was causing contention with other lines of execution essentially robbing the provider of the time allowed to execute when under load.